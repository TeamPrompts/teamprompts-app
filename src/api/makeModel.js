const AUTHOR = 'Author';
const EXAMPLES = 'Examples';
const FITB = 'FITB';
export const ID = 'ID';
const PROMPTS = 'Prompts';

function makeModel(record) {
  const examples = record.get(EXAMPLES)
    ? JSON.parse(record.get(EXAMPLES))
    : undefined;
  const prompts = record.get(PROMPTS)
    ? JSON.parse(record.get(PROMPTS))
    : undefined;
  return {
    author: record.get(AUTHOR),
    examples: examples,
    id: record.get(ID),
    prompts: prompts,
    source: record.get(FITB)
  };
}

export default makeModel;
