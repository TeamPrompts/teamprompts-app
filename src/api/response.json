{
  "records": [
    {
      "id": "recQ2jud83D1dCxr4",
      "fields": {
        "ID": 12,
        "Status": "active",
        "Examples": "[\"to add steps to the workflow\", \"more customers completing the existing workflow\"]",
        "Author": "John Cutler",
        "FITB": "To feel more confident about our decision to __BLANK__, I would like to observe __BLANK__.",
        "PromptValidation": 1,
        "ExampleValidation": 1
      },
      "createdTime": "2019-11-20T17:13:08.000Z"
    },
    {
      "id": "recrYoglrdF32klOv",
      "fields": {
        "ID": 11,
        "Status": "active",
        "Examples": "[\"spend too long deliberating over the behavior of the page filter\",\" test early with customers\"]",
        "Author": "John Cutler",
        "FITB": "I’m concerned that if we __BLANK__, we may miss the opportunity to __BLANK__.",
        "PromptValidation": 1,
        "ExampleValidation": 1
      },
      "createdTime": "2019-11-20T17:11:22.000Z"
    },
    {
      "id": "recNsahDzKCx0IQPk",
      "fields": {
        "ID": 10,
        "Status": "active",
        "FITB": "You’ve got to think of these kinds of __BLANK__ as a way to win,” he said at a press conference on Friday. “How expansive can you __BLANK__ and at the same time get __BLANK__? I want you to expect that we are really managing that.”",
        "PromptValidation": 1,
        "ExampleValidation": 1
      },
      "createdTime": "2019-11-19T18:29:56.000Z"
    },
    {
      "id": "recm6xCbOJSJZ6jwY",
      "fields": {
        "ID": 8,
        "Status": "active",
        "Prompts": "[\"an assumption\",\"that assumption\",\"a list of ways\"]",
        "Author": "John Cutler",
        "FITB": "We seem to be treating __BLANK__ as a foregone conclusion. I’m not sure it is that simple. If we were proven wrong (or less right) about __BLANK__ , it would impact us in the following ways: __BLANK__  .",
        "PromptValidation": 1,
        "ExampleValidation": 1
      },
      "createdTime": "2019-11-17T22:37:01.000Z"
    },
    {
      "id": "recqeN7TDq4kdhFah",
      "fields": {
        "ID": 4,
        "Status": "active",
        "Examples": "[\"this amazing challenge\"]",
        "Author": "John Doe",
        "FITB": "To even know where to start, we would need to better understand __BLANK__.",
        "PromptValidation": 1,
        "ExampleValidation": 1
      },
      "createdTime": "2019-11-17T04:24:19.000Z"
    },
    {
      "id": "rec99C9lxMl6aSedq",
      "fields": {
        "ID": 3,
        "Status": "active",
        "Examples": "[\"this huge problem we have\",\"feel the wrath of all that has come before\"]",
        "Author": "The Person",
        "FITB": "A danger of not resolving __BLANK__ now is that we will eventually __BLANK__.",
        "PromptValidation": 1,
        "ExampleValidation": 1
      },
      "createdTime": "2019-11-17T03:37:43.000Z"
    },
    {
      "id": "reckkQSV8NuHKOAHr",
      "fields": {
        "ID": 2,
        "Status": "active",
        "Prompts": "[\"some assumption\",\"reason\"]",
        "Examples": "[\"how we perceive user onboarding\",\"our projections might be off\"]",
        "Author": "Mary Tine",
        "FITB": "The weakest link in our mental model is __BLANK__. If we’re off -- even by a little -- it could have a big impact on this initiative because __BLANK__.",
        "PromptValidation": 1,
        "ExampleValidation": 1
      },
      "createdTime": "2019-11-17T03:37:43.000Z"
    },
    {
      "id": "receM4Gus7HH43oLM",
      "fields": {
        "ID": 1,
        "Status": "active",
        "Prompts": "[\"an action\",\"reason\",\"new event\"]",
        "Author": "John Cutler",
        "FITB": "We may be over-confident in our ability to __BLANK__. It hasn’t really been a problem yet because __BLANK__, but it could become a problem in the future if __BLANK__.",
        "PromptValidation": 1,
        "ExampleValidation": 1
      },
      "createdTime": "2019-11-17T03:37:43.000Z"
    }
  ]
}
