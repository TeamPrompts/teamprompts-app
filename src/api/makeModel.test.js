import makeModel from './makeModel';
import { records } from './response';

function makeRecord(element) {
  return {
    get: field => element.fields[field]
  };
}

test('models to match snapshot', () => {
  const models = records.map(makeRecord).map(makeModel);

  expect(models).toMatchSnapshot();
});
