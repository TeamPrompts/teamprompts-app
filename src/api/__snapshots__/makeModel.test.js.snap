// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models to match snapshot 1`] = `
Array [
  Object {
    "author": "John Cutler",
    "examples": Array [
      "to add steps to the workflow",
      "more customers completing the existing workflow",
    ],
    "id": 12,
    "prompts": undefined,
    "source": "To feel more confident about our decision to __BLANK__, I would like to observe __BLANK__.",
  },
  Object {
    "author": "John Cutler",
    "examples": Array [
      "spend too long deliberating over the behavior of the page filter",
      " test early with customers",
    ],
    "id": 11,
    "prompts": undefined,
    "source": "I’m concerned that if we __BLANK__, we may miss the opportunity to __BLANK__.",
  },
  Object {
    "author": undefined,
    "examples": undefined,
    "id": 10,
    "prompts": undefined,
    "source": "You’ve got to think of these kinds of __BLANK__ as a way to win,” he said at a press conference on Friday. “How expansive can you __BLANK__ and at the same time get __BLANK__? I want you to expect that we are really managing that.”",
  },
  Object {
    "author": "John Cutler",
    "examples": undefined,
    "id": 8,
    "prompts": Array [
      "an assumption",
      "that assumption",
      "a list of ways",
    ],
    "source": "We seem to be treating __BLANK__ as a foregone conclusion. I’m not sure it is that simple. If we were proven wrong (or less right) about __BLANK__ , it would impact us in the following ways: __BLANK__  .",
  },
  Object {
    "author": "John Doe",
    "examples": Array [
      "this amazing challenge",
    ],
    "id": 4,
    "prompts": undefined,
    "source": "To even know where to start, we would need to better understand __BLANK__.",
  },
  Object {
    "author": "The Person",
    "examples": Array [
      "this huge problem we have",
      "feel the wrath of all that has come before",
    ],
    "id": 3,
    "prompts": undefined,
    "source": "A danger of not resolving __BLANK__ now is that we will eventually __BLANK__.",
  },
  Object {
    "author": "Mary Tine",
    "examples": Array [
      "how we perceive user onboarding",
      "our projections might be off",
    ],
    "id": 2,
    "prompts": Array [
      "some assumption",
      "reason",
    ],
    "source": "The weakest link in our mental model is __BLANK__. If we’re off -- even by a little -- it could have a big impact on this initiative because __BLANK__.",
  },
  Object {
    "author": "John Cutler",
    "examples": undefined,
    "id": 1,
    "prompts": Array [
      "an action",
      "reason",
      "new event",
    ],
    "source": "We may be over-confident in our ability to __BLANK__. It hasn’t really been a problem yet because __BLANK__, but it could become a problem in the future if __BLANK__.",
  },
]
`;
